File: public/docs/_examples/component-styles/dart/lib/hero_app_component.dart
import 'package:angular2/core.dart';
import 'hero.dart';
import 'hero_app_main_component.dart';

// #docregion
@Component(
    selector: 'hero-app',
    template: '''
      <h1>Tour of Heroes</h1>
      <hero-app-main [hero]="hero"></hero-app-main>''',
    styles: const ['h1 { font-weight: normal; }'],
    directives: const [HeroAppMainComponent])
class HeroAppComponent {
// #enddocregion
  Hero hero =
      new Hero('Human Torch', ['Mister Fantastic', 'Invisible Woman', 'Thing']);

  @HostBinding('class')
  String get themeClass => 'theme-light';
// #docregion
}

File: public/docs/_examples/component-styles/ts/app/hero-app.component.ts
import {Component, HostBinding} from '@angular/core';
import {Hero} from './hero';
import {HeroAppMainComponent} from './hero-app-main.component';

// #docregion
@Component({
  selector: 'hero-app',
  template: `
    <h1>Tour of Heroes</h1>
    <hero-app-main [hero]=hero></hero-app-main>`,
  styles: ['h1 { font-weight: normal; }'],
  directives: [HeroAppMainComponent]
})
// #enddocregion
export class HeroAppComponent {
  hero = new Hero(
    'Human Torch',
    ['Mister Fantastic', 'Invisible Woman', 'Thing']
  )
  
  @HostBinding('class') get themeClass() {
    return 'theme-light';
  }

}
